### PR TITLE
feat: initial version

### PR DESCRIPTION
**Pull request by pjenvey**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+/target
+**/*.rs.bk
+Cargo.lock
+.idea
```
Comment (jrconlin) [inline .gitignore]: May want to add 
```
.floo*
/rls
```
(We might want to come up with a common skeleton of these, come to think of it.)

**Comment by pjenvey** [comment]
> missing LICENSE file?

it's in initial commit to master

**Comment by jrconlin** [comment]
Ah! sorry, missed that. Thought this was the first push.

File: .gitignore (hunk 1 above)
Comment (bbangert) [reply .gitignore]: Global .gitignore is ideal for these, as we should only be excluding things the build system or language drop in there that shouldn't be there. IDE specific things that shouldn't be included are user-specific, and should be excluded using global .gitignore (such as that .idea dir).

File: src/aws.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+pub fn get_ec2_instance_id() -> Option<&'static str> {
+    EC2_INSTANCE_ID.as_ref().map(String::as_ref)
+}
+
+fn _get_ec2_instance_id() -> reqwest::Result<String> {
+    let client = reqwest::Client::builder()
+        .timeout(Duration::from_secs(1))
+        .build()?;
+    client
+        .get("http://169.254.169.254/latest/meta-data/instance-id")
```
Comment (kitcambridge) [inline src/aws.rs]: Oh, I remember this from the PushGo days... 😄

**Pull request merged**